bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/DeviceDeserializer.java

```diff
@@ -0,0 +1,35 @@
+package com.bugsnag.android;
+
+import java.util.Map;
+
+class DeviceDeserializer implements MapDeserializer<DeviceWithState> {
+
+    @Override
+    public DeviceWithState deserialize(Map<String, Object> map) {
+        DeviceBuildInfo buildInfo = new DeviceBuildInfo(
+                MapUtils.<String>getOrThrow(map, "manufacturer"),
+                MapUtils.<String>getOrThrow(map, "model"),
+                MapUtils.<String>getOrThrow(map, "osVersion"),
+                MapUtils.<Integer>getOrThrow(map, "apiLevel"),
+                MapUtils.<String>getOrThrow(map, "osBuild"),
+                MapUtils.<String>getOrThrow(map, "fingerprint"),
+                MapUtils.<String>getOrThrow(map, "tags"),
+                MapUtils.<String>getOrThrow(map, "brand"),
+                MapUtils.<String[]>getOrThrow(map, "cpuAbis")
+        );
+
+        String time = MapUtils.getOrNull(map, "time");
+        return new DeviceWithState(
+                buildInfo,
+                buildInfo.getCpuAbis(),
```

Comment:
Why does `Device` have cpuAbis? (I realise this is pre-existing, but seems odd to be in `DeviceBuildInfo` too.)

Reply:
There doesn't appear to be a reason. I've updated the constructor to remove this parameter and populate it from `buildInfo` instead.